File: Sample Code/2_TimingExample/timing.py
# This program demonstrates usage of some timing functions.
# See https://docs.python.org/3.7/library/time.html for more details.

import time

# Monotonic is one of the several timing functions that gets the current time.
# The actual value returned by monotonic isn't useful by itself.
# Instead, you should only be finding the difference in time between two 
# calls to monotonic.
t1 = time.monotonic()

print("Current time:")
print("{0}s".format(t1))

# Wait 500 ms.
time.sleep(0.5)

# Get current time again.
t2 = time.monotonic()
msElapsed = (t2 - t1) * 10 ** 3 # Convert seconds to ms

print("500 ms later:")
print("{0}s, {1:.4}ms elapsed".format(t2, msElapsed))

# Wait 100 us. The Pi will most likely end up waiting significantly longer.
time.sleep(0.0001)

# Get current time again.
t3 = time.monotonic()
usElapsed = (t3 - t2) * 10 ** 6 # Convert seconds to us

print("100 us later:")
print("{0}s, {1:.4}us elapsed".format(t3, usElapsed))

File: Sample Programs (NEW!!!)/cpp/2_TimingExample/timing.cpp
// This program demonstrates usage of some timing functions.
// See http://wiringpi.com/reference/timing/ for more details.

#include <iostream>
#include <wiringPi.h>

using namespace std;

int main()
{
	unsigned int ms1, ms2, ms3;
	unsigned int us1, us2, us3;
	
    // Initializes Wiring Pi and resets the time counter.
    wiringPiSetupGpio();
    
    // Get current time (since Wiring Pi setup was called) in ms and us.
	ms1 = millis();
	us1 = micros();
    
	cout << "Current time:" << endl;
	cout << ms1 << " ms, " << us1 << " us" << endl << endl;
    
    // Wait 500 ms.
	delay(500);
	
    // Get current time again.
	ms2 = millis();
	us2 = micros();
    
	cout << "500 ms later:" << endl;
	cout << ms2 << " ms, " << us2 << " us";
    cout << " (" << (ms2 - ms1) << "ms elapsed, " << (us2 - us1) << "us elapsed)";
    cout << endl << endl;
	
    // Wait 100 us. The Pi will most likely end up waiting significantly longer.
	delayMicroseconds(100);
	
    // Get current time again.
	ms3 = millis();
	us3 = micros();
    
	cout << "100 us later:" << endl;
	cout << ms3 << " ms, " << us3 << " us";
    cout << " (" << (ms3 - ms2) << "ms elapsed, " << (us3 - us2) << "us elapsed)";
    cout << endl << endl;
}
